DoSomethingArchive/legacy-website
# Fact collection field collection

Refs #2674

@sergii-tkachenko Would you mind reviewing?
## How to test
- Pull down staging DB
- Run the following commands (which is the same order that our `ds build` script executes in as well):
  - `drush updb` -- Should run `dosomething_fact_page_update_7002`
  - `drush -y fr dosomething_fact_page`
- Confirm your local Fact Page nodes display the same Facts as they do on staging
- Confirm the instance of `field_facts`, a multi-value Entityreference field, has been removed from the `fact_page` contentt ype.
## Overview
- Creates a field collection, `field_fact_collection`, with fields:
  - Fact (entityreference to the Fact node)
  - Copy
  - Campaign
- Migrates the Fact Page's `field_facts` values into the new `field_fact_collection`, into the `field_fact` field.
- Alters `dosomething_fact_page_preprocess_node` to use a new method, `dosomething_fact_page_get_field_fact_collection_vars`, which inspects the field collection data
- If copy field is set, appends the copy to the end of the fact. Links to the campaign (if set) whereever an editor uses a [link] token.
- Deletes the `field_facts` instance from the Fact Page node
## Screenshot

![screen shot 2014-07-08 at 9 53 47 am](https://cloud.githubusercontent.com/assets/1236811/3510543/a1ca4baa-06a7-11e4-9fa4-b56d64446ffc.png)
![screen shot 2014-07-08 at 9 56 00 am](https://cloud.githubusercontent.com/assets/1236811/3510541/9dbb6f26-06a7-11e4-8c1f-1cb0c66bd17d.png)
